refinery/data_set_manager/views.py

```diff
@@ -1059,11 +1060,21 @@ def post(self, request):
             return HttpResponseForbidden()
 
         logger.debug("Adding files to data set '%s'", data_set)
-        for file_store_item in data_set.get_file_store_items():
-            if (file_store_item.source.startswith(
-                    (settings.REFINERY_DATA_IMPORT_DIR, 's3://')
-            )):
-                import_file.delay(file_store_item.uuid)
+        for node in data_set.get_nodes():
+            file_store_item = node.get_file_store_item()
+            if file_store_item is not None:
+                if not file_store_item.datafile.name:
+                    if (file_store_item.source.startswith(
+                        (settings.REFINERY_DATA_IMPORT_DIR, 's3://')
+                    )):
+                        # Put Node into PENDING state in the UI by removing
+                        # the associated FileStoreItem's import_task_id and
+                        # updating said Node's Solr index entry
+                        file_store_item.import_task_id = ""
+                        file_store_item.save()
+                        NodeIndex().update_object(node,
```

Comment:
Yes, that is exactly the problem. Since we are retrieving this information from Solr, without an `update_object` here, a PENDING state will never be returned and the datafile will remain a lightning bolt until that `import_file` completes.

Reply:
I see. Why does `PENDING` state depend on `import_task_id` being blank?